### PR TITLE
added slim version of misty

### PR DESCRIPTION
**Pull request by schae211**
first version of misty-py.

### CODE REVIEW

File: liana/method/sp/_misty.py [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+        missing_targets = set(targets).difference(set(adata_target.var_names))
+        if len(missing_targets) > 0:
+            logging.warning(f"Missing target features: {missing_targets}.")
+        targets = list(set(targets).difference(missing_targets))
+    else:
+        targets = adata_target.var_names.tolist()
+    return(targets)
+
+
+def _check_predictors(adata_source, predictors):
```
Comment (dbdimitrov) [inline liana/method/sp/_misty.py:60]: same as `_check_targets`

File: liana/method/sp/_misty.py [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+                                             add_self=add_self, spatial_key=spatial_key, zoi=zoi)
+    return(views)
+
+
+def _check_anndata_objects_groups(xdata, ydata, spatial_key, group_intra_by, group_env_by):
+    # check that the adata_source and adata_target have the same number of spots/cells
+    assert xdata.X.shape[0] == ydata.X.shape[0]
+
+    # check that the adata_source and adata_target have the same spatial coordinates
+    assert np.all(xdata.obsm[spatial_key] == ydata.obsm[spatial_key])
```
[the remark this answers is not in the document]
Comment (dbdimitrov) [reply liana/method/sp/_misty.py:121]: (also applies to my local functions)

File: liana/method/sp/_misty.py [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+    
+    # check the group_target_by and group_views_by variables
+    if group_intra_by is not None:
+        assert group_intra_by in ydata.obs.columns
+    if group_env_by is not None:
+        assert group_env_by in xdata.obs.columns
+
+
+def _check_target_in_predictors(target, predictors):
+    # TODO: this is only necessary for the intra view, for the other views we could make it optional
```
Comment (dbdimitrov) [inline liana/method/sp/_misty.py:131]: to clarify with @schae211

Comment (dbdimitrov) [reply liana/method/sp/_misty.py:131]: add `self_interactions` param with False as default - show example where self_interactions are of interest - e.g. cell types in labelled high res spatial

Comment (schae211) [reply liana/method/sp/_misty.py]: This feature is implemented now, see parameter 'keep_same_predictor' (feel free to rename it)

File: liana/method/sp/_misty.py (hunk 1 above)
Comment (schae211) [reply liana/method/sp/_misty.py]: Now there is a single function called `_check_features`.

**Pull request merged**